crates/libcontainer/src/container/builder.rs

```diff
@@ -1,18 +1,18 @@
 use crate::error::{ErrInvalidID, LibcontainerError};
-use crate::workload::default::DefaultExecutor;
-use crate::workload::{Executor, ExecutorManager};
-use crate::{syscall::Syscall, utils::PathBufExt};
+use crate::syscall::syscall::SyscallType;
+use crate::utils::PathBufExt;
+use crate::workload::{self, Executor};
 use std::path::PathBuf;
 
 use super::{init_builder::InitContainerBuilder, tenant_builder::TenantContainerBuilder};
 
-pub struct ContainerBuilder<'a> {
+pub struct ContainerBuilder {
     /// Id of the container
     pub(super) container_id: String,
     /// Root directory for container state
     pub(super) root_path: PathBuf,
     /// Interface to operating system primitives
-    pub(super) syscall: &'a dyn Syscall,
+    pub(super) syscall: SyscallType,
```

Comment:
Since this is a breaking change for the users of libcontainer, please create another PR.

Reply:
I will make these changes into a series of PR so we can these changes out.